users/user_permission.go

```diff
@@ -1,96 +1,74 @@
 package users
 
 import (
-	"bytes"
-	"encoding/json"
+	"context"
 	"fmt"
-	"io/ioutil"
 	"net/http"
 
+	"github.com/SKF/go-rest-utility/client"
+	"github.com/go-http-utils/headers"
 	"github.com/pkg/errors"
 )
 
-func AddUserRole(identityToken, stage, userID, role string) (err error) {
-	user, err := getUser(identityToken, stage, userID)
+func AddUserRole(identityToken, stage, userID, role string) error {
+	return AddUserRoleWithContext(context.Background(), identityToken, stage, userID, role)
+}
+
+func AddUserRoleWithContext(ctx context.Context, identityToken, stage, userID, role string) (err error) {
+	user, err := getUser(ctx, identityToken, stage, userID)
 	if err != nil {
 		return
 	}
 
 	user.UserRoles = append(user.UserRoles, role)
-	return updateUser(identityToken, stage, user)
+	return updateUser(ctx, identityToken, stage, user)
 }
 
-func getUser(identityToken, stage, userID string) (user user, err error) {
+func getUser(ctx context.Context, identityToken, stage, userID string) (user user, err error) {
 	if userID == "" {
 		return user, fmt.Errorf("userID is required")
 	}
 
-	url := fmt.Sprintf(accessMgmtBaseURL+"/users/%s", stage, userID)
-	req, err := http.NewRequest(http.MethodGet, url, nil)
-	if err != nil {
-		err = errors.Wrap(err, "http.NewRequest failed")
-		return
-	}
-
-	req.Header.Set("Authorization", identityToken)
-	req.Header.Set("Content-Type", "application/json")
+	req := client.Get("/users/{id}").
+		Assign("id", userID).
+		SetHeader(headers.ContentType, "application/json")
 
-	client := &http.Client{}
-	resp, err := client.Do(req)
+	rest := httpClientAccessMgmt(stage, identityToken)
```

Comment:
why "rest" and not "client" or "restClient"?

Reply:
Would be nice if you could explain why you don't like the naming "rest".
I will change it to "restClient"

Reply:
"rest" doesn't mean client and this is a client, the name should reflect what it is/what it's used for